docs/ldml/tr35-numbers.md

```diff
@@ -386,36 +386,74 @@ A pattern `type` attribute is used for _compact number formats_, such as the fol
+        <pattern type="10000" count="one" alt="alphaNextToNumber">¤ 00K</pattern>
+        <pattern type="10000" count="other">¤00K</pattern>
+        <pattern type="10000" count="other" alt="alphaNextToNumber">¤ 00K</pattern>
+        <pattern type="100000" count="one">¤000K</pattern>
+        <pattern type="100000" count="one" alt="alphaNextToNumber">¤ 000K</pattern>
+        <pattern type="100000" count="other">¤000K</pattern>
+        <pattern type="100000" count="other" alt="alphaNextToNumber">¤ 000K</pattern>
+        <pattern type="1000000" count="one">¤0M</pattern>
+        <pattern type="1000000" count="one" alt="alphaNextToNumber">¤ 0M</pattern>
         …
     </currencyFormat>
 </currencyFormatLength>
 ```
 
 Formats can be supplied for numbers (as above) or for currencies or other units. They can also be used with ranges of numbers, resulting in formatting strings like “$10K” or “$3–7M”.
 
-To format a number N, the greatest type less than or equal to N is used, with the appropriate plural category. N is divided by the type, after removing the number of zeros in the pattern, less 1. APIs supporting this format should provide control over the number of significant or fraction digits.
+To format a number N, use the following steps. The examples use N = 123456, the currency = CAD, and the currency symbol string = "$CA"
+1. Let P be the pattern element with greatest type less than or equal to N, and any count value.
+    * P = `<pattern type="100000" count="**one**">¤000K</pattern>`
+2. Let V be the pattern element value.
+    * V = "¤000K" 
+3. If the element value of P is "0", then use the corresponding non-compact number formatting instead, and skip the rest of these steps — but adjust the precision as described below.
+    * For example, instead of `currencyFormat` `<pattern type="10000" count="one">¤00K</pattern>`, use `<pattern>¤#,##0.00</pattern>`.
+5. If P is a currency format, look at the currency symbol string, and the position of the currency symbol ¤ in the pattern element value.
+If ¤ is immediately to the left of a 0 and the currency string ends with a _letter grapheme cluster_ (eg, $CA),
+or to the right and the currency starts with a _letter grapheme cluster_ (eg, CA$),
+then switch to the `alt=alphaNextToNumber` pattern.
+    * P = `<pattern type="100000" count="**one**" alt="alphaNextToNumber">¤ 000K</pattern>` // with the currency symbol CA$
+    * V = "¤ 000K" 
+6. Let Z be the number of 0 characters in V, minus 1.
+    * Z = 2
+7. Let V' be the same as V, but replacing that sequence of zeros by "{0}".
+    * V' = "¤ {0}K"
+8. Let T be the numeric value of the `type` attribute value, after removing the final Z zeros.
+    * "100000" removing "00" = "1000"
+    * T  = 1000
+9. Let N' be N / T
+    * N = 123.456
+10. Determine the plural category of N, based on the numeric precision settings (the min/max number of significant or fraction digits), and switch the pev if necessary.
```

Comment:
What's a "pev"?

Reply:
Sorry, V used to be pev, and I didn't catch that one.